frontstage/views/account/accept_transfer_survey.py

```diff
@@ -0,0 +1,119 @@
+        # Handle api errors
+        if exc.status_code == 409:
+            logger.info('Expired transfer survey email verification token', token=token, api_url=exc.url,
+                        api_status_code=exc.status_code)
+            abort(409)
+        elif exc.status_code == 404:
+            logger.warning('Unrecognised transfer survey email verification token', token=token, api_url=exc.url,
+                           api_status_code=exc.status_code)
+            abort(404)
+        else:
+            logger.info('Failed to verify transfer survey email', token=token, api_url=exc.url,
+                        api_status_code=exc.status_code)
+            raise exc
+
+
+@account_bp.route('/confirm-transfer-surveys/<batch>', methods=['GET'])
+def accept_transfer_surveys(batch):
+    """
+    Accept endpoint when a transfer survey summary is accepted
+    :param batch: batch number
+    :type batch: str
+    """
+    logger.info('Attempting to get batch number', batch_number=batch)
+    try:
+        response = party_controller.get_pending_surveys_batch_number(batch)
+        is_existing_user = _is_existing_account(response.json()[0]['email_address'])
+    except ApiError as exc:
+        logger.error('Failed to confirm transfer survey', status=exc.status_code, batch_number=batch)
+        raise exc
+    if is_existing_user:
+        return redirect(url_for('account_bp.accept_transfer_surveys_existing_account', batch=batch))
+    return redirect(url_for('register_bp.pending_surveys_register_enter_your_details', batch_no=batch,
+                            email=response.json()[0]['email_address'], is_transfer=True))
+
+
+@account_bp.route('/confirm-transfer-surveys/<batch>/existing-account', methods=['GET'])
+@jwt_authorization(request)
+def accept_transfer_surveys_existing_account(session, batch):
+    """
+    Accept redirect endpoint for accepting transfer surveys for existing account
+    :param session:
+    :type session:
+    :param batch: batch number
+    :type batch: str
+    """
+    logger.info('Attempting to confirm transfer surveys for existing account', batch_number=batch)
+    try:
+        party_controller.confirm_pending_survey(batch)
+    except ApiError as exc:
+        logger.error('Failed to confirm transfer survey for existing account', status=exc.status_code, batch_number=batch)
+        raise exc
+    logger.info('Successfully completed transfer survey for existing account', batch_number=batch)
+    return render_template('surveys/surveys-transfer/transfer-survey-complete-thank-you.html')
+
+
+def _is_existing_account(respondent_email):
+    """
+    Checks if the respondent already exists against the email address provided
+    :param respondent_email: email of the respondent
+    :type respondent_email: str
+    :return: returns true if account already registered
+    :rtype: bool
+    """
+    respondent = party_controller.get_respondent_by_email(respondent_email)
+    if not respondent:
```

Comment:
Very minor, but we could just `return respondent is not None`

Reply:
🎨

Reply:
this is a validation check function, either we do here or else we will have to do the check at the calling place. I believe it ok what you say?

Reply:
It's fine, it's just a code style thing to turn it into a 1 line check instead of 3